request_test.go

```diff
@@ -352,31 +352,46 @@ func TestRequest(t *testing.T) {
 }
 
 func TestHandleHTTPHEADRequest(t *testing.T) {
-	modifierFunc := func(m *RequestModifier) {
-		m.Header.Set("foo", "bar")
-	}
-	w := utils.NewWriterStub()
-	req := &http.Request{Method: "HEAD", Header: make(http.Header)}
 	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprint(w, "Hi")
 		st.Expect(t, r.Header.Get("foo"), "")
 	})
-	interceptor := Request(modifierFunc)
+	interceptor := Request(func(m *RequestModifier) {
+		m.Header.Set("foo", "bar")
+	})
+	w := utils.NewWriterStub()
+	req := &http.Request{Method: "HEAD", Header: make(http.Header)}
 	interceptor.HandleHTTP(w, req, handler)
 	st.Expect(t, string(w.Body), "Hi")
 }
 
 func TestHandleHTTPOPTIONSRequest(t *testing.T) {
```

Comment:
I would rename test functions into `TestHandleHTTPOptionsRequest`. 
I might be just a personal style preference, but for me the HTTP verb name is upper case mostly inside the HTTP protocol domain & spec, not strictly required in program semantics, such as methods/function names. 
`HTTP` keyword instead is an acronym and the idiomatic Go also points about keeping acronyms with original camel case notations.